src/Emulator/Extensions/Mocks/HPSHostController.cs

```diff
@@ -171,6 +174,40 @@ private byte[] GetBytesFromSlave(int count, TimeInterval timeInterval)
             return result;
         }
 
+        // This method is for getting OA1EN, 
+        // returns true when OA1EN is enabled.
+        private bool OA1ENEnabled(){
+            return currentDoubleSlave.ReadDoubleWord(0x08) >> 15 != 0x00;
+        }
+
+        // This method is for getting RXEN,
+        // returns true when the buffer is empty,
+        private bool RXNECleared(){
+            return (currentDoubleSlave.ReadDoubleWord(0x18) & (1 << 2)) == 0x00;
+        }
+        
+        private delegate bool RegisterBitAccessDeligate();
+
+        private void PollForRegisterBit(RegisterBitName bitName){
+            int i = 0;
+            RegisterBitAccessDeligate registerAccess;
+            switch(bitName){
+                case RegisterBitName.OA1EN:
+                    registerAccess = OA1ENEnabled;
+                    break;
+                case RegisterBitName.RXNE:
+                    registerAccess = RXNECleared;
+                    break;
+                default:
+                    registerAccess = null;
+                    break;
+            }
+            while(!registerAccess() && i < 8){
```

Comment:
Why 8? Can't we poll forever here?

Reply:
In the future there's a socket limitation of 5s waiting time. I guess there are better solutions than this.

Reply:
If you want to wait for up to 5s in total it's better to get start time, and compare whether the delta is over 5s, so the intent would be clearer.

Reply:
As per https://github.com/renode/renode-infrastructure/pull/70#discussion_r918318115
I could change it to stopwatch implementation again if that's better.